torch/csrc/inductor/aoti_runtime/arrayref_tensor.h

```diff
@@ -0,0 +1,365 @@
+#pragma once
+
+#include <torch/csrc/inductor/aoti_runtime/model.h>
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+
+#include <assert.h>
+#include <cstdint>
+#include <cstring>
+
+namespace torch {
+namespace aot_inductor {
+
+// Can't use c10::ArrayRef because it's not truly header-only and
+// pulls in other c10 headers. This is (sadly) copy-pasted and
+// adapted.
+template <typename T>
+class MiniArrayRef final {
```

Comment:
I don't have strong preference, but am wondering if it would be better to place arrayref_tensor into aoti_torch instead of aoti_runtime?